src/P3_processes.jl

```diff
@@ -99,3 +99,130 @@ function ice_melt(
+
+# Arguments
+- `state`: [`P3State`](@ref)
+- `aps`: [`CMP.AirProperties`](@ref) 
+- `tps`: `TDP.ThermodynamicsParameters`
+- `velocity_params`: velocity parameterization, e.g. [`CMP.Chen2022VelType`](@ref)
+- `ρₐ`: air density [kg/m³]
+- `Tₐ`: air temperature [K]
+
+This rate represents the thermodynamic upper limit to collisional freezing, 
+which occurs when the heat transfer from the ice particle to the environment is 
+balanced by the latent heat of fusion.
+
+From Eq (A7) in Musil (1970), [Musil1970](@cite).
+"""
+function compute_max_freeze_rate(state, aps, tps, velocity_params, ρₐ, Tₐ)
+    (; D_vapor, K_therm) = aps
+    cp_l = TDP.cp_l(tps)
+    T_frz = TDP.T_freeze(tps)
+    Lᵥ = TD.latent_heat_vapor(tps, Tₐ)
+    L_f = TD.latent_heat_fusion(tps, Tₐ)
+    Tₛ = T_frz  # the surface of the ice particle is assumed to be at the freezing temperature
+    ΔT = Tₛ - Tₐ  # temperature difference between the surface of the ice particle and the air
+    Δρᵥ_sat =
+        ρₐ * (  # saturation vapor density difference between the surface of the ice particle and the air
+            TD.q_vap_saturation(tps, Tₛ, ρₐ, TD.PhaseNonEquil) -
+            TD.q_vap_saturation(tps, Tₐ, ρₐ, TD.PhaseNonEquil)
+        )
+    v_term = ice_particle_terminal_velocity(state, velocity_params, ρₐ)
+    F_v = CO.ventilation_factor(state.params.vent, aps, v_term)
+    function max_freeze_rate(Dᵢ)
+        Tₐ ≥ T_frz && return zero(Dᵢ)  # No collisional freezing above the freezing temperature
+        return 2 * (π * Dᵢ) * F_v(Dᵢ) * (K_therm * ΔT + Lᵥ * D_vapor * Δρᵥ_sat) / (L_f - cp_l * ΔT)
+    end
+    return max_freeze_rate
+end
+
+"""
+    compute_local_rime_density(state, velocity_params, ρₐ, T)
+
+Provides a function `ρ′_rim(Dᵢ, Dₗ)` that computes the local rime density [kg/m³] 
+    for a given ice particle diameter `Dᵢ` [m] and liquid particle diameter `Dₗ` [m].
+
+# Arguments
+- `state`: [`P3State`](@ref)
+- `velocity_params`: velocity parameterization, e.g. [`CMP.Chen2022VelType`](@ref)
+- `ρₐ`: air density [kg/m³]
+- `T`: temperature [K]
+
+# Returns
+A function that computes the local rime density [kg/m³] using the equation:
+
+```math
+ρ'_{rim} = 51 + 114 R_i - 5.5 R_i^2
+```
+where
+```math
+R_i = \\frac{ D_{liq} ⋅ |v_{liq} - v_{ice}| }{ 2 T_{sfc} }
+```
+and ``T_{sfc}`` is the surface temperature [°C], ``D_{liq}`` is the liquid particle
+diameter [m], ``v_{liq/ice}`` is the particle terminal velocity [m/s].
+So the units of ``R_i`` are [m² s⁻¹ K⁻¹]. The units of ``ρ'_{rim}`` are [kg/m³].
+
+They assume for simplicity that ``T_{sfc}`` equals ``T``, the ambient air temperature.
+For real graupel, ``T_{sfc}`` is slightly higher than ``T`` due to latent heat release 
+of freezing liquid particles onto the ice particle. Morrison & Milbrandt (2013) 
+found little sensitivity to "realistic" increases in ``T_{sfc}``.
+
+Implementation follows Cober and List (1993), Eq. 16 and 17.
```

Comment:
If I understand the reference paper correctly, Eq. 16 represents `R_i` as a function of ambient velocity. However, Eq. 14 represents `R_i` as a function of impact velocity (which could be given by `v_liq - v_ice`). But if you use Eq. 14 then for the rime density Eq. 15 must be used (which has different coefficients).